### PR TITLE
[CRAW-411] added counters for long hosts

### PR DESCRIPTION
**Pull request by gd-smalygin**

### CODE REVIEW

**Comment by agatash-ias** [comment]
Looks good :+1: Are unit tests required to fully complete?

File: src/java/org/apache/nutch/crawl/GeneratorReducer.java [hunk 1]
```diff
@@ -71,10 +71,15 @@ protected void reduce(SelectorEntry key, Iterable<WebPage> values,
           hostCountMap.put(hostordomain, 0);
           hostCount = 0;
         }
+        hostCountMap.put(hostordomain, hostCount + 1);
         if (hostCount >= maxCount) {
-          return;
+          if (hostCount == maxCount) {
+            context.getCounter("Generator", "LONG_HOST").increment(1);
+          }
+          context.getCounter("Generator", "LONG_HOST_URL_CUT").increment(1);
```
Comment (MichealKum) [inline src/java/org/apache/nutch/crawl/GeneratorReducer.java]: Change the for-loop above using separate iterator to aviod expensive domain name parsing after the limit is reached. The tail of the values should be read here. And increment the counter only once, use local varable to measuer the tail.

Comment (MichealKum) [inline src/java/org/apache/nutch/crawl/GeneratorReducer.java]: On second thought, usually there is only one item in the iterator in our crawler, so it does not worth to make this changes.

**Comment by MichealKum** [comment]
:+1:

**Pull request closed without merging**